src/details/ArborX_Ray.hpp

```diff
@@ -250,19 +257,27 @@ KOKKOS_INLINE_FUNCTION float overlapDistance(Ray const &ray,
   float const a1 = 2.f * dotProduct(ray.direction(), oc);
   float const a0 = dotProduct(oc, oc) - r * r;
 
-  float t1;
-  float t2;
-  if (!solveQuadratic(a2, a1, a0, t1, t2))
+  if (solveQuadratic(a2, a1, a0, tmin, tmax))
   {
-    // No intersection of a bidirectional ray with the sphere
-    return 0.f;
+    // ensures that tmin <= tmax
+    if (tmin > tmax)
+      KokkosExt::swap(tmin, tmax);
+
+    return true;
   }
-  float tmin = KokkosExt::min(t1, t2);
-  float tmax = KokkosExt::max(t1, t2);
+  constexpr auto inf = KokkosExt::ArithmeticTraits::infinity<float>::value;
+  tmin = inf;
+  tmax = -inf;
+  return false;
+}
 
-  if (tmax < 0)
+KOKKOS_INLINE_FUNCTION float overlapDistance(Ray const &ray,
```

Comment:
Make the overlapDistance function general for both ray/sphere and ray/box cases?

Reply:
Can do but really didn't want to write more tests

Reply:
This function might be simple enough to test lightly. We can even leave it to the user side.
But if we keep it, it feels strange to have it for ray/sphere but not ray/box.

Reply:
I had to implement it mainly for backward compatibility.  I haven't made up my mind yet whether we should provide it or not.  I would prefer if we discussed that in follow up work